Public/Src/App/Bxl/Args.cs

```diff
@@ -290,6 +290,12 @@ public bool TryParse(string[] args, PathTable pathTable, out ICommandLineConfigu
                         OptionHandlerFactory.CreateBoolOption(
                             "debugScript",
                             opt => frontEndConfiguration.DebugScript = opt),
+                        OptionHandlerFactory.CreateOption(
+                            "globalPassthroughEnvironmentVariables",

```

Comment:
>globalPassthroughEnvironmentVariables [](start = 29, length = 37)

I know this was discussed at a design meeting I missed, but should we add 'unsafe' to the arguments? Since in the pip api we clearly mark these as unsafe as well.

Reply:
This is actually more unsafe than the one on the api because that one salts the cache, this one even bypasses a cache fingerprint and will cache regardless of these settings.

---
In reply to: [292244405](https://github.com/microsoft/BuildXL/pull/442#discussion_r292244405) [](ancestors = 292244405)

Reply:
> globalPassthroughEnvironmentVariables [](start = 29, length = 37)
> 
> I know this was discussed at a design meeting I missed, but should we add 'unsafe' to the arguments? Since in the pip api we clearly mark these as unsafe as well.

Do you mean just the naming? Or I need to do special thing to make it "unsafe"?

Reply:
Just naming. yes

Reply:
> Just naming. yes

Thanks, I will change it

Reply:
All of our unsafe options start with "unsafe_".

---
In reply to: [292249068](https://github.com/microsoft/BuildXL/pull/442#discussion_r292249068) [](ancestors = 292249068)